src/pudl/transform/vceregen.py

```diff
@@ -0,0 +1,206 @@
+"""Transformations of the Vibrant Clean Energy renewable generation profiles.
+
+Wind and solar profiles are extracted separately, but concatenated into a single table
+in this module, as they have exactly the same structure.
+"""
+
+import pandas as pd
+from dagster import asset
+
+import pudl
+from pudl.helpers import cleanstrings_snake, simplify_columns, zero_pad_numeric_string
+from pudl.metadata.dfs import POLITICAL_SUBDIVISIONS
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def _prep_lat_long_fips_df(raw_vcegen__lat_lon_fips: pd.DataFrame) -> pd.DataFrame:
+    """Prep the lat_long_fips table to merge into the capacity factor tables.
+
+    Prep entails making sure the formatting and column names match those in the
+    capacity factor tables, adding 0s to the beginning of FIPS codes with 4 values,
+    and making separate county and state columns.
+    """
+    logger.info(
+        "Preping Lat-Long-FIPS table for merging with the capacity factor tables"
+    )
+    # Get state name patterns from the POLITICAL_SUBDIVISIONS table
+    ps_usa_df = POLITICAL_SUBDIVISIONS[POLITICAL_SUBDIVISIONS["country_code"] == "USA"]
+    state_names = cleanstrings_snake(
+        ps_usa_df, ["subdivision_name"]
+    ).subdivision_name.tolist()
+    state_pattern = "|".join(state_names)
```

Comment:
We could also just literally use the state and county _values_ in the `POLITICAL_SUBDIVISIONS` df....but I guess I felt like I should use the ones in this table? If that feels silly then lmk.

Reply:
The political subdivisions only has state level information, so we can't get the county names there.

But given the FIPS county code, we can look up the county name and state in the Census DP1 data, or (probably?) using `addfips` though there's a chance that a handful of the values would be different or some of the FIPS codes wouldn't show up, since they do evolve slightly over time. You might try merging in the county names and states from DP1 and see if they all match up as expected.

The only thing we really need the `County_State` strings for is associating the FIPS code with each column of the original data. If we have the FIPS code, that's the authoritative ID, and the county name is really just for human readability. The state will be useful for geographic filtering. We use the 2-letter state abbreviation for state columns elsewhere and they're easier to work with than the long state names. And those you should definitely be able to look up in the political subdivision dataframe given the first 2-digits of the FIPS code.

Reply:
I can't decide what to do about the `county` name column. Should I keep the county name as extracted from this VCE FIPS table or should I use the county name from the census dp1 table? I think @e-belfer mentioned that the census table was older and possibly less accurate than their provided FIPS/county combos.

Reply:
I'm inclined to just use the county name from the VCE fips table (instead of the dp1 table) because of how the VCE table actually has multiple county names per FIPS code. As I mention above in relation to the primary keys, I'm not sure why, but that's all to say that mapping to the dp1 table by FIPS code would be inaccurate.

Reply:
That makes sense!